presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -345,8 +347,23 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
                     if (immutablePartitions) {
                         throw new PrestoException(HIVE_PARTITION_READ_ONLY, "Unpartitioned Hive tables are immutable");
                     }
-                    updateMode = UpdateMode.APPEND;
-                    writeInfo = locationService.getTableWriteInfo(locationHandle);
+
+                    if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.APPEND) {
+                        updateMode = UpdateMode.APPEND;
+                        writeInfo = locationService.getTableWriteInfo(locationHandle);
+                    }
+                    else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.OVERWRITE) {
+                        updateMode = UpdateMode.OVERWRITE;
+                        writeInfo = locationService.getTableWriteInfo(locationHandle, true);
+                    }
+                    else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ERROR) {
+                        throw new PrestoException(HIVE_TABLE_READ_ONLY,
+                                "Cannot insert into an readonly unpartitioned Hive table.");
+                    }
+                    else {
+                        throw new IllegalArgumentException(format("Unsupported insert existing table behavior: %s",
```

Comment:
Nit: don't wrap here. Also, if there is only one argument at the end, use string concatenation:
```java
throw new IllegalArgumentException("Unsupported insert existing table behavior: " + insertExistingPartitionsBehavior);